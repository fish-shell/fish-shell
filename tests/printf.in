printf "Hello %d %i %f %F %g %G\n" 1 2 3 4 5 6

printf "%x %X %o %llu\n" 10 11 8 -1
# %a has OS-dependent output - see #1139
#printf "%a %A\n" 14 15

printf "%c %s\n" a hello
printf "%c%c%c\n" hello â€¦ o
printf "%e %E\n" 5 6

printf "%20d\n" 50
printf "%-20d%d\n" 5 10

printf "%*d\n" 10 100

printf "%%\"\\\n" 
printf "%s\b%s\n" x y
printf "abc\rdef\n"
printf "Msg1\fMsg2\n"
printf "foo\vbar\vbaz\n"
printf "\111 \x50 \u0051 \U00000052"

echo
echo "Test escapes"

# \c escape means "stop printing"
printf 'a\cb'
echo

# Bogus printf specifier, should produce no stdout
printf "%5" 10 ^ /dev/null

true
