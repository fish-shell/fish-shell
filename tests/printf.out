Hello 1 2 3.000000 4.000000 5 6
a B 10 18446744073709551615
0xep+0 0XFP+0
a hello
5.000000e+00 6.000000E+00
                  50
5                   10
       100
%"\nxy
abcdef
Msg1Msg2
foobarbaz
I P Q R
Test escapes
a
